### PR TITLE
Option to create Docker image and push to Quay

### PR DESCRIPTION
**Pull request by coverbeck**
Updated the shared GitHub actions deploy workflow to optionally create a docker image and push it to quay.

Part of work for SEAB-6653. We'll create a Docker image from dockstore-support using this; the run a container from the image to do the topic updating.

**Review Instructions**

1. After the corresponding dockstore-support PR has been merged
2. Create a tag in dockstore-support
3. Run the deploy_workflow action in dockstore_support
4. After it completes, look at https://quay.io/repository/dockstore/dockstore-support?tab=tags to see if an image was created.

### CODE REVIEW

File: .github/workflows/deploy_tagged.yaml [hunk 1]
```diff
@@ -7,13 +7,31 @@ on:
         description: 'patch and pre-release metadata'
         required: true
         default: '.0-alpha.1'
+      dockerize:
```
Comment (denis-yuen) [inline .github/workflows/deploy_tagged.yaml]: Don't use the verb "dockerize"
I think we use the jwilder project dockerize https://github.com/dockstore/dockstore/blob/ebe3be2bd3d0353b192cbcd564de4d94cc2e45e0/.circleci/config.yml#L601 

(Assuming this is unrelated)

**Review by kathy-t — APPROVED** [review]

**Review by david4096 — APPROVED** [review]

**Review by svonworl — APPROVED** [review]
